warehouse/tunnelling/model.go

```diff
@@ -0,0 +1,72 @@
+package tunnelling
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	stunnel "github.com/rudderlabs/sql-tunnels/driver/ssh"
+)
+
+var ErrMissingKey = errors.New("missing mandatory key")
+var ErrUnexpectedType = errors.New("unexpected type")
+
+const (
+	SSHForward Type = "ssh_forward"
+)
+
+type Type string
+type Config map[string]interface{}
+
+type TunnelInfo struct {
+	Config Config
+}
+
+func ReadSSHTunnelConfig(config map[string]interface{}) (conf *stunnel.Config, err error) {
+
+	var (
+		user, host, port, privateKey *string
+	)
+
+	if user, err = ReadString("sshUser", config); err != nil {
```

Comment:
Can we use constants for sshUser, sshHost, sshPort, sshPrivateKey?

Reply:
Done.